### PR TITLE
Add shinytest2

### PR DESCRIPTION
**Pull request by jr-nicola**
- Set up {shinytest2} to work with app
- Add a basic test to check contents of navbar are as expected. (More tests can be added later as the app structure changes)

Depends on #8 being merged first

### CODE REVIEW

File: tests/testthat/test-app-function.R [hunk 1]
```diff
@@ -0,0 +1,11 @@
+test_that("tfpbrowser app initial values are consistent", {
```
Comment (russHyde) [inline tests/testthat/test-app-function.R:1]: This test fails for me locally

```r
Error in `$<-.data.frame`(`*tmp*`, "call_text", value = c("shinytest2::AppDriver$new(shiny_app, name = \"tfpbrowser\")",  : 
  replacement has 9 rows, data has 7
```

It also won't run in non-interactive setting, so the tests won't be usable in CI. Is this the intended setup?

**Review by russHyde — COMMENTED** [review]
Thanks for this, I'm not sure what to suggest at present - the new test throws an error for me locally, and the interactive() restriction means they won't run in CI. If you're happy for the devs to run tests outside of CI, then I'm fine with okaying the code. I'm really not sure what caused the test to fail.


I haven't any experience with shinytest2 yet, although it looks a great improvement on shinytest. Could you confirm whether shinytest2 tests play nicely when non-shinytest testthat code is present in the same directory, and also, whether you think a testfile prefix might be useful to distinguish shinytest2-files from testthat/testServer-files.

File: tests/testthat/test-app-function.R (hunk 1 above)
Comment (jr-nicola) [reply tests/testthat/test-app-function.R]: I've renamed the file to test-shinytest2.R to distinguish it from other {testthat} tests we may want to include (which should work side-by-side with shinytest2). 

Running `testthat::test_local()` locally doesn't give me that error, which is slightly odd. I get one warning saying that there are non-unique HTML values for treeview (this should be fixed in the convert-to-modules MR coming soon) One possible issue - shinytest2 requires that your package to be installed when testing, was the most recent version of {tfpbrowser} installed before running tests?

Removing the `interactive()` statement broke R CMD check locally (`'Shiny app did not become stable in 15000ms'`) so I'm okay with running it locally (for now)

**Review by russHyde — APPROVED** [review]
LGTM

**Pull request merged**